actions/views/browser.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {GenericAction} from 'mattermost-redux/types/actions';
+
+import {Constants, ActionTypes, WindowSizes} from 'utils/constants';
+
+export function emitBrowserWindowResized(): GenericAction {
+    const width = window.innerWidth;
+
+    let windowSize;
+    switch (true) {
+    case width > Constants.DESKTOP_SCREEN_WIDTH: {
```

Comment:
Total nitpicking, but I'm still not a fan of using `switch(true)` instead of an if-else statement. I just don't think it adds enough to justify introducing something new that other devs might have to learn

Reply:
IMO it's a totally valid and arguably better way of writing conditional logic in situations with multiple potential options and should be something JS developers should know about. No time like the present to learn!

What is interesting to me is the varying opinions around the different ways of writing conditional logic in JS at MM. I would bet that if you put 5 developers in a room and asked them to present their preferred approach, you would get 5 different approaches. :P Every time I go to write something with conditional logic, I find my self questioning what approach to use that might minimize change requests and pretty much each time someone requests a change. 🤷‍♂️

In the absence of a definitive style guide on how we prefer to use conditional logic at MM or at least linting that catches for situations we would prefer not to use, personal preference is going to win at the end of the day so long is it is technically correct. This is one of mine personal preferences. :)

Reply:
this discussion is being held often and in many variations. Some might say that the `if/else if/else` approach adds readability or more clear intention, but IMO the `switch (true)` style is the better approach as well.

This is not only a matter of personal preference: there are several tests out there in the wild that prove that `switch` is computed faster than `if/else`

Reply:
Also when looking for some objective reason to argue around, I made a quick benchmark to compare these. I tried to write it so that stuff doesn't get optimized out by the JS compiler, so hopefully I didn't write something that favours one over the other, but if-else does seem to be a tiny bit faster. That's on the scale of tens of microseconds on my machine though, so it's too small to matter either way. Guess we're stuck with personal preference here then lol.